rai/node/node.cpp

```diff
@@ -200,7 +200,7 @@ bool confirm_block (MDB_txn * transaction_a, rai::node & node_a, T & list_a, std
 				blake2b_update (&hash, address.bytes.data (), address.bytes.size ());
 				uint16_t peer_delay;
 				blake2b_final (&hash, &peer_delay, sizeof (peer_delay));
-				peer_delay &= (1 << 10) - 1; // maximum of 1023 milliseconds delay
+				peer_delay &= (1 << 7) - 1; // maximum of 127 milliseconds delay
```

Comment:
This edit made me wonder if the lower and upper limits should be configurable, in case there turns out be problems in either direction (speed vs security) ? Default limits, overridable from config.json.

Something like:

`peer_delay = boost::algorithm::clamp (peer_delay, min_delay, max_delay)`

Reply:
`clamp` wouldn't work, as the delay would be way more likely to be either the min or the max. However, it's probably a good idea to let users configure a power of 2 for the max.

Reply:
@PlasmaPower but isn't clamping to max basically what you're doing anyway with `peer_delay &= (1 << 7) - 1;`  ?

Reply:
No the `&=` means it will be uniformly distributed across the range 0 to 123 (assuming blake2 has a uniform distribution).

The lowest 7 bits will be kept, and the others will be discarded. If, say, the 8th bit was set, it wouldn't clamp it to 123, but instead just unset the bit and keep the other bits as they are (resulting in a random number between 0 and 123).

Reply:
@PlasmaPower Yeah, I see that now, clamping would be completely wrong. Thanks for explaining.